polaris-react/src/components/Tag/Tag.scss

```diff
@@ -147,22 +107,31 @@ $button-size: 20px;
   @include unstyled-button;
   display: block;
   flex-shrink: 0;
-  height: $button-size;
-  width: $button-size;
-  margin: 0 var(--p-space-1);
-  border-radius: var(--p-border-radius-full);
+  height: 18px;
+  width: 18px;
+  margin: var(--p-space-025);
+  margin-left: var(--p-space-05);
+  // stylelint-disable-next-line polaris/border/declaration-property-unit-disallowed-list -- override border-radius
+  border-radius: 7px;
+  color: var(--p-color-icon-subdued);
 
   svg {
-    fill: var(--p-color-icon);
+    fill: currentColor;
```

Comment:
Had to change this to `currentColor` due to specificity issues on close button for disabled tags.